utils/metric_tracking.py

```diff
@@ -89,3 +120,64 @@ def save(self, overwrite=True):
                 os.remove(self.path)
 
         torch.save(self.metrics, self.path)
+
+    def collect_per_epoch(self):
+        epoch_metrics = {"epochs": []}
+        for k, _ in self.metrics_to_track.items():
+            epoch_metrics["{}_mean".format(k)] = []
+            epoch_metrics["{}_std".format(k)] = []
+
+        epochs = self.metrics["epochs"]
+        unique_epochs = np.unique(epochs)
+        epoch_metrics["epochs"] = unique_epochs
+
+        for k, v in self.metrics.items():
+            if k in self.metrics_to_track:
+                v = np.array(v)
+                if k != "iterations" and k != "epochs":
+                    for this_epoch in unique_epochs:
+                        where_metrics = epochs == this_epoch
+                        v_mean = np.mean(v[where_metrics])
+                        v_std = np.std(v[where_metrics])
+                        epoch_metrics["{}_mean".format(k)].append(v_mean)
+                        epoch_metrics["{}_std".format(k)].append(v_std)
+        return epoch_metrics
+
+    def get_best_epoch_for_metric(self, metric_name, evaluation_metric=np.argmax):
+        return evaluation_metric(self.collect_per_epoch()[metric_name])
+
```

Comment:
I think having a single line function which just calls another function is a little dodgy. Better to just do this directly from wherever this is supposed to be called?